### PR TITLE
batches: change condition for closable changesets

### PR DESCRIPTION
**Pull request by BolajiOlajide**
When calculating available bulk operations for changesets, we activate the `Close Changeset` operation on changesets that are in a failed state.

However, in this case, it shouldn't be because the changesets might not have been published, so it's impossible to close it.

## Test plan

* Manual testing
* Updated unit tests

### CODE REVIEW

**Review by fkling — APPROVED** [review]

File: internal/batches/service/service.go [hunk 1]
```diff
@@ -1708,7 +1708,7 @@ func (s *Service) GetAvailableBulkOperations(ctx context.Context, opts GetAvaila
 
 		// can changeset be published
 		isChangesetCommentable := isChangesetOpen || isChangesetDraft || isChangesetMerged || isChangesetClosed
-		isChangesetClosable := isChangesetOpen || isChangesetDraft || isChangesetJobFailed
+		isChangesetClosable := isChangesetOpen || isChangesetDraft
```
Comment (eseliger) [inline internal/batches/service/service.go:1711]: I think the best check would be `isChangesetClosable := changeset.ExternalState != ""`, WDYT?

**Review by eseliger — APPROVED** [review]

File: internal/batches/service/service.go (hunk 1 above)
Comment (BolajiOlajide) [inline internal/batches/service/service.go:1711]:
```suggestion
		isChangesetClosable := changeset.ExternalState != ""
```

Comment (eseliger) [reply internal/batches/service/service.go:1711]: in that case, should we check for externalState IN (open, draft)?

Comment (BolajiOlajide) [reply internal/batches/service/service.go:1711]: We do.

Have a [look here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@bo/change-conds-closable-changeset/-/blob/internal/batches/service/service.go?L1702%3A3-1703%3A82)

Comment (eseliger) [reply internal/batches/service/service.go:1711]: god, sorry

**Pull request merged**